### PR TITLE
DAOS-7796 dfuse: Fix a bug with file modes on open.

### PR DESCRIPTION
**Pull request by ashleypittman**
When write-back caching is on and file is opened in write-only
mode dfuse is changing the file mode to allow reading, however
this was being incorrectly applied so dfs was rejecting it.
Clear the bits before adding read mode.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: utils/node_local_test.py [hunk 1]
```diff
@@ -1342,6 +1342,46 @@ def test_cache(self):
...
+        dfuse1.start(v_hint='two_1')
+
+        file0 = os.path.join(dfuse0.dir, 'file')
+        fd = open(file0, 'w')
+        fd.write('test')
+        fd.close()
+
+        file1 = os.path.join(dfuse1.dir, 'file')
+        fd = open(file1, 'r')
+        data = fd.read()
```
Comment (daltonbohning) [inline utils/node_local_test.py:1372]: Should `data` here be the same as was written?
Locally, this is what I see:
```
$ mkdir dfuse1 && dfuse -m dfuse1
$ mkdir dfuse2 && dfuse -m dfuse2
$ echo test > dfuse1/$POOL/$CONT/file && cat dfuse2/$POOL/$CONT/file
test
$ echo updated > dfuse1/$POOL/$CONT/file && cat dfuse2/$POOL/$CONT/file
test
```

**Review by daltonbohning — COMMENTED** [review]
I'm seeing the EINVAL issue fixed locally - just had one comment about read/write consistency

File: utils/node_local_test.py (hunk 1 above)
Comment (ashleypittman) [reply utils/node_local_test.py:1372]: Yes it should, I just didn't test for that.

Comment (ashleypittman) [reply utils/node_local_test.py:1377]: Done.

Comment (daltonbohning) [reply utils/node_local_test.py:1377]: What about on the case of update? (like with the echo + cat example above)
I'm genuinely asking - not sure if it's expected to be in-sync
For example:
```
        // First write
        file0 = os.path.join(dfuse0.dir, 'file')
        fd = open(file0, 'w')
        fd.write('test')
        fd.close()

        // First read. dfuse1 does not yet have this cached
        file1 = os.path.join(dfuse1.dir, 'file')
        fd = open(file1, 'r')
        data = fd.read()
        fd.close()
        print(data)
        assert data == 'test'

        // Second write to same file
        fd = open(file0, 'w')
        fd.write('updated')
        fd.close()

        // Second read. dfuse1 has this cached from the first read
        fd = open(file1, 'r')
        data = fd.read()
        fd.close()
        print(data)
        assert data == 'updated'
```

Comment (daltonbohning) [reply utils/node_local_test.py:1377]: Makes sense, thanks!

**Review by daltonbohning — APPROVED** [review]
LGTM. Fixes the issue reported in DAOS-7796

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**